### PR TITLE
Scrolling object collection enable/disables renderers instead of gameobjects

### PR DESCRIPTION
**Pull request by RogPodge**
## Overview
Scrolling object collection now disables/enables renderers when the object leaves the clippingbounds rather than toggling the entire renderer gameobject. 

@vaoliva I noticed we actually had this behavior before, is there a reason why it was changed to toggle gameobjects rather than just the component?

https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8430/files#diff-16544eaccc0f696813d8291c3acc7afdfec625c42395585f6c08a81a31d75c74L1608-R1667

## Changes
- Fixes: #9121

### CODE REVIEW

File: Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 1]
```diff
@@ -1651,9 +1651,9 @@ private void ManageVisibility(bool isRestoringVisibility = false)
             {
                 if (clippedRenderer != null && !clippedRenderer.transform.IsChildOf(ScrollContainer.transform))
                 {
-                    if (!clippedRenderer.gameObject.activeSelf)
+                    if (!clippedRenderer.enabled)
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:1654]: Tested the example scene out and found that the objects in this container did not render after a scroll.  The behavior also appears to be changed for the other items in the scrolling object collection examples.

| Scroll Before | Scroll Current|
|---|---|
| ScrollingObjBefore |ScrollingObjAfter

Comment (RogPodge) [reply Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:1654]: Seems to be an issue with those prefabs being pre-initialized with those out of view game objects being disabled, so when the script is run currently, those objects never get enabled again 😅. Not super sure what the best way to address this while avoiding breaking the current experience for our users.

Comment (RogPodge) [reply Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs]: Added new performance options that will let users toggle between the behaviors

image

File: Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 2]
```diff
@@ -845,6 +845,35 @@ public enum PaginationMode
             ToCellIndex // To selected cell
         }
 
+        #region performance variables
+        [SerializeField]
+        [Tooltip("Disables Gameobjects with Renderer components which are clipped by the clipping box.")]
+        private bool disableClippedGameobjects = true;
+
+        /// <summary>
+        /// Disables Gameobjects with Renderer components which are clipped by the clipping box.
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:854]: A note about of the performance implications for each option would be helpful.

File: Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 3]
```diff
@@ -845,6 +845,35 @@ public enum PaginationMode
...
+        /// </summary>
+        public bool DisableClippedGameobjects
+        {
+            get { return disableClippedGameobjects; }
+            set { disableClippedGameobjects = value; }
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:859]: nit: capitalize the O in `DisableClippedGameObjects`

File: Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 4]
```diff
@@ -1651,9 +1680,19 @@ private void ManageVisibility(bool isRestoringVisibility = false)
             {
                 if (clippedRenderer != null && !clippedRenderer.transform.IsChildOf(ScrollContainer.transform))
                 {
-                    if (!clippedRenderer.gameObject.activeSelf)
+                    if(disableClippedGameobjects)
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:1683]:
```suggestion
                    if (disableClippedGameobjects)
```

File: Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 5]
```diff
@@ -1651,9 +1680,19 @@ private void ManageVisibility(bool isRestoringVisibility = false)
...
                     {
-                        clippedRenderer.gameObject.SetActive(true);
+                        if (!clippedRenderer.gameObject.activeSelf)
+                        {
+                            clippedRenderer.gameObject.SetActive(true);
+                        }
+                    }
+                    if(disableClippedRenderers)
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/ScrollingObjectCollection/ScrollingObjectCollection.cs:1690]:
```suggestion
                    if (disableClippedRenderers)
```

**Review by CDiaz-MS — APPROVED** [review]

**Pull request merged**